# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/IDBCursor.cpp
# Commit: d2772a7482ed
# Full Hash: d2772a7482ed9394a2c30acc6d03caa36317e75f
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-09-30 16:21:37
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Allow current key to be provided with PBackgroundIDBCursor::Continue. r=ttung,asuth
#   
#   The key is assumed to be unset for now.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D43251
# ==============================================================================

diff -r 340dfa69aed8 -r d2772a7482ed dom/indexedDB/IDBCursor.cpp
--- a/dom/indexedDB/IDBCursor.cpp	Fri Sep 27 10:11:45 2019 +0000
+++ b/dom/indexedDB/IDBCursor.cpp	Fri Sep 27 10:11:41 2019 +0000
@@ -449,7 +449,7 @@
         IDB_LOG_STRINGIFY(key));
   }
 
-  mBackgroundActor->SendContinueInternal(ContinueParams(key));
+  mBackgroundActor->SendContinueInternal(ContinueParams(key), Key());
 
   mContinueCalled = true;
 }
@@ -555,7 +555,7 @@
       IDB_LOG_STRINGIFY(key), IDB_LOG_STRINGIFY(primaryKey));
 
   mBackgroundActor->SendContinueInternal(
-      ContinuePrimaryKeyParams(key, primaryKey));
+      ContinuePrimaryKeyParams(key, primaryKey), Key());
 
   mContinueCalled = true;
 }
@@ -600,7 +600,7 @@
         IDB_LOG_STRINGIFY(mSourceIndex), IDB_LOG_STRINGIFY(mDirection), aCount);
   }
 
-  mBackgroundActor->SendContinueInternal(AdvanceParams(aCount));
+  mBackgroundActor->SendContinueInternal(AdvanceParams(aCount), Key());
 
   mContinueCalled = true;
 }